### ISSUE TITLE
Interface Changes (major release change)

### ISSUE THREAD

**Issue by tracker1** [pioardi/poolifier#6]
Suggestions... 

(documentation update) for the worker, you don't even need to extend the class, there's no real value here.  Remove the class definition and change the exports to...

    module.exports = new ThreadWorker(yourFunction, {...})

(interface change, major update) For the pool, put the emitter on the pool itself, you can extend your initial FixedThreadPool/DynamicThreadPool with the emitter... you can add the emitter to your prototype chain or base class(es), The eventemitter methods will also work if you Object.assign(YourClass.prototype, EventEmitter.prototype); and add EventEmitter.call(this) inside your constructor.

For the pool, instead of passing event handler options, utilize the event emitter...

    pool.on('error', error => ...);
    pool.on('ready', () => ...);

You may also want a method that returns a promise bound to your ready event...  This will make waiting for the pool to be ready easier to utilize.  This will mean a little complexity internally to resolving the ready state...

    // fixed and dynamic thread pool...
    class FixedThreadPool extends EventEmitter {
    ...

    __ready = false; // track ready state

    constructor() {
      // inheritance methods
      super();

      // set ready state listener for ready event
      this.once('ready', () => this.__ready = true);

      // initialize the pool - existing functionality
      ...
    }

    // ready method for convenience
    async ready() {
      if (__ready) return this; // already ready
      return new Promise((resolve, reject) => {
        this.once('ready', () => resolve(this));
        this.once('error', reject); // only rejects if error before ready
      });
    }

    // static method, better convenience
    static create(...args) { // match constructor args
      return (new FixedThreadPool(...args)).ready();
    }

Then, in practical use, you can do...

    const pool = await (new FixedThreadPool(...)).ready();
    // or even better...
    const pool = await FixedThreadPool.create(...);

Cool abstraction concept btw.

**Comment by tracker1** [comment]
@pioardi fyi, I updated my initial description to include some more details (eventemitter in constructor, and static create method).

**Comment by pioardi** [comment]
Hi @tracker1  ,
thanks for your help , so my thoughts : 

1) module.exports = new ThreadWorker(yourFunction, {...}) , **you are right**  thanks to catch this
2a ) About the event emitter you are also right , I can do FixedThreadPool extends...EventEmitter 
2b) Ready state , yes this will be cool and also will be possible to remove some tricks on the unit tests
3) Will be a great developer experience to have this const pool = await FixedThreadPool.create(...);

So basically I agree on this , now how we want to proceed ? 
Would you like to build this by yourself and submit a PR?

Let me know and thanks !!! 💯

**Comment by tracker1** [comment]
I'm afraid I really don't have the time... I'm on some pretty agressive timelines at work right now... just figured I'd make the suggestions so that your library is even easier to use.

**Comment by pioardi** [comment]
Thank you @tracker1  I will do the changes :)
Are you planning to use the library ?

**Comment by pioardi** [comment]
@tracker1  I completed changes related to some of your feedbacks .
I do not changed the event emitter ( this will not have any benefits on developer experience ) 
and for now the "ready" state of a worker thread seems handled by nodejs at low level .
I changed examples as you suggested and the destroy method to await thread termination :) 

Thanks again , you are welcome in this project :)
